src/test/java/com/algolia/search/integration/client/SecuredAPIKeyTest.java

```diff
@@ -69,5 +52,9 @@ void testSecuredAPIKey() throws Exception {
 
     assertThatThrownBy(() -> index2WithRestriction.search(new Query()))
         .hasMessageContaining("Index not allowed with this API key");
+
+    clientWithRestriction.close();
```

Comment:
Same as below :) as we don't have the `@AfterAll` anymore we could use a try with resources to close the client?

Reply:
Agreed. I will update the commits.

Reply:
done